tests/ui_tools/test_boxes.py

```diff
@@ -743,74 +743,105 @@ def test_generic_autocomplete_user_mentions(
+            ("#g", 1, None, {}),
+            ("#Stream 1", 0, "#**Stream 1**", {}),  # Complete match.
+            ("#nomatch", 0, None, {}),
+            ("#ene", 0, None, {}),
             # Complex autocomplete prefixes.
-            ("[#Stream", 0, "[#**Stream 1**", []),
-            ("(#Stream", 1, "(#**Stream 2**", []),
-            ("@#Stream", 0, "@#**Stream 1**", []),
-            ("@_#Stream", 0, "@_#**Stream 1**", []),
-            (":#Stream", 0, ":#**Stream 1**", []),
-            ("##Stream", 0, "##**Stream 1**", []),
-            ("##*Stream", 0, None, []),  # NOTE: Optional single star fails
-            ("##**Stream", 0, "##**Stream 1**", []),  # Optional 2-stars
+            ("[#Stream", 0, "[#**Stream 1**", {}),
+            ("(#Stream", 1, "(#**Stream 2**", {}),
+            ("@#Stream", 0, "@#**Stream 1**", {}),
+            ("@_#Stream", 0, "@_#**Stream 1**", {}),
+            (":#Stream", 0, ":#**Stream 1**", {}),
+            ("##Stream", 0, "##**Stream 1**", {}),
+            ("##*Stream", 0, None, {}),  # NOTE: Optional single star fails
+            ("##**Stream", 0, "##**Stream 1**", {}),  # Optional 2-stars
             # With 'Secret stream' pinned.
             (
                 "#Stream",
                 0,
                 "#**Secret stream**",
-                ["Secret stream"],
+                {"pinned": ["Secret stream"]},
             ),  # 2nd-word startswith match (pinned).
             (
                 "#Stream",
                 1,
                 "#**Stream 1**",
-                ["Secret stream"],
+                {"pinned": ["Secret stream"]},
             ),  # 1st-word startswith match (unpinned).
             (
                 "#Stream",
                 2,
                 "#**Stream 2**",
-                ["Secret stream"],
+                {"pinned": ["Secret stream"]},
             ),  # 1st-word startswith match (unpinned).
             (
                 "#Stream",
                 3,
                 "#**Some general stream**",
-                ["Secret stream"],
+                {"pinned": ["Secret stream"]},
+            ),  # 3rd-word starstwith match (unpinned).
+            (
+                "#Stream",
+                4,
+                "#**Web public stream**",
+                {"pinned": ["Secret stream"]},
             ),  # 3rd-word starstwith match (unpinned).
             # With 'Stream 1' and 'Secret stream' pinned.
-            ("#Stream", 0, "#**Stream 1**", ["Secret stream", "Stream 1"]),
-            ("#Stream", 1, "#**Secret stream**", ["Secret stream", "Stream 1"]),
-            ("#Stream", 2, "#**Stream 2**", ["Secret stream", "Stream 1"]),
-            ("#Stream", 3, "#**Some general stream**", ["Secret stream", "Stream 1"]),
+            (
+                "#Stream",
+                0,
+                "#**Stream 1**",
+                {"pinned": ["Secret stream", "Stream 1"]},
+            ),  # 1st-word startswith match (pinned).
```

Comment:
You seem to have added these comments? Without them the test cases fit on one line still?